### DISCUSSION TITLE
When folder is deleted, fsearch doesn't update immediately

### DISCUSSION THREAD

**Discussion by hitbuyi** [cboxdoerfer/fsearch#474] [Q&A]
updating speed is kinda of slow, compared to everything

**Comment by cboxdoerfer** [comment]
File system monitoring (and updating search results immediately when a file gets deleted) is already being worked on and will be part of the next release.

As for the updating speed, if you're referring to the time it takes to do a full (re)scan of the database, this will never be as fast as Everything. Everything is reading the MFT of the NTFS file system to build its database. This is a rather cheap operation, because only a single file must be accessed and parsed. Linux file system (ext4, xfs, btrfs, ...) don't have anything comparable to a MFT, so FSearch has to traverse the whole file system tree to build its database; this is an expensive operation and even with fast NVME drives is much slower than parsing a MFT.
